File: Queue/Python/queue.py
MAX_Q_SIZE = 100

#Class used for queue implementation
class MyQueue():
    def __init__(self, capacity=MAX_Q_SIZE):
        self.front = self.size = 0
        self.rear = capacity - 1
        self.Q = [None] * capacity
        self.capacity = capacity

    def IsEmpty(self):
        return self.size == 0

    def IsFull(self):
        return self.capacity == self.size

    def EnQueue(self, item):
        if self.IsFull():
            print("Full")
            return
        self.rear = (self.rear + 1) % (self.capacity)
        self.Q[self.rear] = item
        self.size += 1
        print("Enqueued", str(item))

    def DeQueue(self):
        if self.IsEmpty():
            print("Empty")
            return

        print("Dequeued", str(self.Q[self.front]))
        self.front = (self.front + 1) % (self.capacity)
        self.size -= 1

    def Queue_Front(self):
        if self.IsEmpty():
            print("Empty queue")
            return
        print("Front item is", str(self.front))

    def Queue_rear(self):
        if self.IsEmpty():
            print("Empty queue")
            return
        print("Rear item is", str(self.rear))

if __name__ == '__main__':
    q = MyQueue()
    MyQueue.EnQueue(3)

File: Queue/C/queue.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <string.h>

#define MAX_SIZE 10

typedef struct queue
{
	int current_size;
	int head;
	int tail;
	int items[MAX_SIZE];
}myQueue;

//Function to check if the queue is empty
int is_empty(myQueue *queue);

//Function to remove of queue
int dequeue(myQueue *queue);

//Function to add to queue
void enqueue(myQueue *queue, int item);

//Function to start the queue
myQueue* start_queue();

int main()
{
	myQueue *queue = start_queue();
	return 0;
}

//Function to start the queue
myQueue* start_queue()
{
	myQueue *new_queue = (myQueue*)malloc(sizeof(myQueue));
	new_queue->current_size = 0;
	new_queue->head = 0;
	new_queue->tail = MAX_SIZE - 1;
	return new_queue;
}

//Function to add to queue
void enqueue(myQueue *queue, int item)
{
	if (queue->current_size >= MAX_SIZE)
	{
		printf("QUEUE OVERFLOW!\n");
	}
	else
	{
		queue->tail = (queue->tail + 1) % MAX_SIZE;
		queue->items[queue->tail] = item;
		queue->current_size++;
	}
}

//Function to remove of queue
int dequeue(myQueue *queue)
{
	if (is_empty(queue))
	{
		printf("QUEUE UNDERFLOW!\n");
		return -1;
	}
	else
	{
		int dequeued = queue->items[queue->head];
		queue->head = (queue->head + 1) % MAX_SIZE;
		queue->current_size--;
		return dequeued;
	}
}

//Function to check if the queue is empty
int is_empty(myQueue *queue)
{
	if (queue->current_size == 0)
	{
		return 1;
	}
	else
	{
		return 0;
	}
}